vmm/src/vm.rs

```diff
@@ -550,24 +550,28 @@ impl Vm {
         }
 
         if let Some(desired_memory) = desired_memory {
-            if self
-                .memory_manager
+            self.memory_manager
                 .lock()
                 .unwrap()
                 .resize(desired_memory)
-                .map_err(Error::MemoryManager)?
-            {
-                self.device_manager
-                    .lock()
-                    .unwrap()
-                    .update_memory()
-                    .map_err(Error::DeviceManager)?;
-
-                self.device_manager
-                    .lock()
-                    .unwrap()
-                    .notify_hotplug(HotPlugNotificationFlags::MEMORY_DEVICES_CHANGED)
-                    .map_err(Error::DeviceManager)?;
+                .map_err(Error::MemoryManager)?;
```

Comment:
Please make sure you don't lose the previous check. It's only if this returned `true` that we would update the memory and notify through ACPI.

Reply:
> Please make sure you don't lose the previous check. It's only if this returned `true` that we would update the memory and notify through ACPI.

@sameo Please take a look on this part.

Reply:
@teawater Can you try and squash https://github.com/sameo/cloud-hypervisor-1/commit/05dfa3e0ee21d183d26d00d9da5333d72b867a28 into 05dfa3e0ee21d183d26d00d9da5333d72b867a28 ?